### PR TITLE
Added tox testing on windows through github action

### PR DESCRIPTION
**Pull request by wndyy**

### CODE REVIEW

File: tests/test_auto_machine.py [hunk 1]
```diff
@@ -41,8 +41,9 @@ def test_auto_machine():
 
     assert config_dict['machine']=="auto"
 
+    send_text = str(platform.system()) + " " + str(platform.release()) + ": This is an automated test message."
```
Comment (cvraut) [inline tests/test_auto_machine.py:44]: I'm seeing a lotta code copy-pasted from test_whecho_simple.py try using imports to lessen the amount of copy-pasted code here.

please revise this file.

File: whecho/_config.py [hunk 2]
```diff
@@ -7,9 +7,13 @@
 import socket
 import platform
 
+try:
```
Comment (cvraut) [inline whecho/_config.py:10]: add comment to this try-except block. Not clear why this exists.

please revise this file

**Review by cvraut — CHANGES_REQUESTED** [review]
looks good. revise `test_auto_machine.py` and `_config.py`

File: .github/workflows/pytest.yml [hunk 3]
```diff
@@ -17,7 +17,10 @@ env:
 
 jobs:
   test:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
```
[the remark this answers is not in the document]
Comment (wndyy) [reply .github/workflows/pytest.yml:20]: awesome

File: tests/test_whecho_simple.py [hunk 4]
```diff
@@ -3,16 +3,18 @@
 import os
 import subprocess
 import re
+import platform
```
[the remark this answers is not in the document]
Comment (wndyy) [reply tests/test_whecho_simple.py:6]: awesome

File: tests/test_auto_machine.py (hunk 1 above)
Comment (wndyy) [reply tests/test_auto_machine.py:44]: send_text is easy to convert to a global variable in test_whecho_simple and import into test_auto_machine

My issue is that there is a lot of other copy-pasted code such as using Popen for running command line

File: whecho/_config.py (hunk 2 above)
Comment (wndyy) [reply whecho/_config.py:10]: getpass.getuser() causes errors when testing on Windows OS. My solution was to hard code "user" as the username for these issues.

Comment (cvraut) [reply whecho/_config.py:10]: is there an alternative to `getpass.getuser()` we can look into?

File: tests/test_auto_machine.py (hunk 1 above)
Comment (cvraut) [reply tests/test_auto_machine.py:44]: let me switch to the branch and take a look at it

Comment (cvraut) [reply tests/test_auto_machine.py]: check commit [fdb8a6f](https://github.com/cvraut/whecho/pull/15/commits/fdb8a6f305d75d0b2f7bfc256ce7dc817d6e970b) for how I re-used code from test_whecho_simple. This file should be good now. Fix the other one too.

File: whecho/_config.py (hunk 2 above)
Comment (wndyy) [reply whecho/_config.py:11]: comment added. not able to find an alternative at the moment.

**Review by cvraut — APPROVED** [review]
changes look good. I think we can approve this into main.

**Comment by cvraut** [comment]
before I merge, does this change any behavior with the package (ie: do I need to update the version on pypi)?

**Comment by wndyy** [comment]
The behavior of the package is the same. I just added testing for Windows. You do not need to update on pypi/

**Pull request merged**